WooCommerce/WooCommerceTests/POS/ViewModels/PointOfSaleDashboardViewModelTests.swift

```diff
@@ -64,13 +68,59 @@ final class PointOfSaleDashboardViewModelTests: XCTestCase {
         // Then
         XCTAssertEqual(sut.isSyncingItems, false)
     }
+
+    func test_isSyncingItems_is_true_when_reload_is_invoked_then_toggled_to_false_when_completed() async throws {
+        XCTAssertEqual(sut.isSyncingItems, true, "Precondition")
+
+        // Given/When
+        await sut.reload()
+
+        // Then
+        XCTAssertEqual(sut.isSyncingItems, false)
+    }
+
+    func test_isSyncingItems_is_true_when_reload_is_invoked_then_toggled_to_false_when_error() async throws {
+        // Given
+        let itemProvider = MockPOSItemProvider()
+        itemProvider.shouldThrowError = true
+
+        let sut = PointOfSaleDashboardViewModel(itemProvider: itemProvider,
+                                                cardPresentPaymentService: cardPresentPaymentService,
+                                                orderService: orderService)
+        XCTAssertEqual(sut.isSyncingItems, true, "Precondition")
+
+        // Given/When
+        await sut.reload()
+
+        // Then
+        XCTAssertEqual(sut.isSyncingItems, false)
+    }
+
+    func test_reload_invokes_providePointOfSaleItems() async {
```

Comment:
Not quite sure how helpful this specific reload test is at the moment, since is mainly testing the mock? I think ideally we can mock this with different simulated network responses (like we do in `POSProductProviderTests`) and observe if the amount of `items` was changed before vs after the reload, but it feels too much setup at the moment, and maybe something we want to simplify 🤔